src/NHibernate/Criterion/LikeExpression.cs

```diff
@@ -78,17 +83,28 @@ public override SqlString ToSqlString(ICriteria criteria, ICriteriaQuery criteri
 			else
 				lhs.Add(columns[0]);
 
+			IType propertyType = projection.GetTypes(criteria,criteriaQuery).Single();
+			if (value != null && !(value is System.Type) && !propertyType.ReturnedClass.IsInstanceOfType(value))
+			{
+				throw new QueryException(string.Format(
+											"Type mismatch in {0}: expected type {1}, actual type {2}",
+											criteria.GetType(), propertyType.ReturnedClass, value.GetType()));
```

Comment:
Here formatting is messy. Please use tab for indenting and SPACE for alignment to avoid it breaking with different tab widths.